crates/bevy_ecs/src/bundle/insert.rs

```diff
@@ -140,28 +140,29 @@ impl<'w> BundleInserter<'w> {
         inserter
     }
 
-    /// # Safety
-    /// `entity` must currently exist in the source archetype for this inserter. `location`
-    /// must be `entity`'s location in the archetype. `T` must match this [`BundleInfo`]'s type
-    #[inline]
-    pub(crate) unsafe fn insert<T: DynamicBundle>(
-        &mut self,
+    unsafe fn before_insert<'a>(
```

Comment:
This and `after_insert` should have comments documenting why they're separate from each other and insert itself.